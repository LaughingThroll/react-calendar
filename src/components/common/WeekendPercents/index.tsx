import React from 'react'
import styles from './weekendPercents.module.scss'

export interface WeekendPercents {
  percent: number
  classNames?: string[]
}

const { weekendPercent } = styles

const WeekendPercentsComponent: React.FC<WeekendPercents> = ({ percent, classNames = [] }) => {
  return <div className={[weekendPercent, ...classNames].join(' ')}>{percent} %</div>
}

export default React.memo(WeekendPercentsComponent)
