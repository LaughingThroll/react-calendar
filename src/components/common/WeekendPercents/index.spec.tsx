import { shallow } from 'enzyme'
import WeekendPercents, { WeekendPercentsProps } from './'

const setUp = (props: WeekendPercentsProps) => shallow(<WeekendPercents {...props} />)

describe('WeekendPercents render', () => {
  it('default render', () => {
    const props = { percent: 15 }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
    expect(tree.text().includes('15')).toBe(true)
  })

  it('render with percent < 0', () => {
    const props = { percent: -10 }
    const tree = setUp(props)

    expect(tree.text().includes('-10')).toBe(true)
  })
})
