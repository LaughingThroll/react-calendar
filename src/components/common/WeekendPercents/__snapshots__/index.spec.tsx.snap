// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekendPercents render default render 1`] = `
<div
  className="weekendPercent"
>
  15
   %
</div>
`;
